### PR TITLE
Lightning: fix incorrect parse tidb release version

### LINKED ISSUE #37007: Lightning: tidb version number cause parallel import conflict records
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Build tidb binary by yourself, and make tidb shows commit id as tidb release version:
```
mysql> select tidb_version();
Release Version: bcf61918
...
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
```

when you setup parallel import, and run it against the given tidb cluster, it will show duplicate records error.

The issue is because https://github.com/pingcap/tidb/blob/release-6.1/br/pkg/lightning/restore/restore.go#L1592 the version number parsing logic will try to match the first version number it meet in the version string, so the tikv min version will be used by lightning as tidb release version. and then, it will skips the logic due the tidb version not large than 4.0. Such behavior will cause duplicate records error.

We need to fix the tidb version parsing logic: https://github.com/pingcap/tidb/blob/release-6.1/br/pkg/version/version.go#L321
We can display some warning when tidb version number can't be found.

### 2. What did you expect to see? (Required)
Import success

### 3. What did you see instead (Required)
import failures due to duplicate records

### 4. What is your TiDB version? (Required)
custom binary

```
mysql> select tidb_version();
Release Version: bcf61918
...
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
```

### PR DESCRIPTION
**Pull request by whitekeepwork**
### What problem does this PR solve?
this PR resolved parallel import conflict caused by unexpected output from tidb_version() return.

Issue Number: close #37007 

Problem Summary:
tidb version number cause parallel import conflict records

### What is changed and how it works?
In the FetchVersion function, I added a precheck for querying the results of the tidb_version() function before it returns.
If querying results didn't match a pattern of ***Release Version: v***, then there are no valid release versions that will be parsed by ParseServerInfo. it should turn into fall back and return the result from ***select version()***

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix Lighting and BR incorrect parsing tidb version.
```

### CODE REVIEW

File: br/pkg/version/version_test.go [hunk 1]
```diff
@@ -536,3 +536,28 @@ Check Table Before Drop: false`
 	require.Error(t, err)
 	require.Regexp(t, "mock failure$", err.Error())
 }
+
+func TestFetchVersionWithCommitID(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	require.NoError(t, err)
+
+	tidbVersionWithCommitInReleaseVersion := `Release Version: bcf61918
```
Comment (D3Hunter) [inline br/pkg/version/version_test.go:544]: please add a positive case for normal `release version`

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/version/version_test.go (hunk 1 above)
Comment (whitekeepwork) [reply br/pkg/version/version_test.go:554]: thanks for commend, the positive case has been added

**Review by D3Hunter — APPROVED** [review]
LGTM

File: br/pkg/version/version_test.go [hunk 2]
```diff
@@ -536,3 +536,44 @@ Check Table Before Drop: false`
 	require.Error(t, err)
 	require.Regexp(t, "mock failure$", err.Error())
 }
+
+func TestFetchVersionWithCommitID(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	require.NoError(t, err)
+
+	tidbVersionWithValideReleaseVersion := `Release Version: v5.2.1
```
Comment (buchuitoudegou) [inline br/pkg/version/version_test.go:544]:
```suggestion
	tidbVersionWithValidReleaseVersion := `Release Version: v5.2.1
```

**Review by buchuitoudegou — APPROVED** [review]
rest lgtm

**Comment by D3Hunter** [comment]
ci pass
https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/13040/pipeline

**Pull request merged**